### PR TITLE
#166325993 Upload image

### PR DESCRIPTION
**Pull request by djallas**
#### What does this PR do?
Upload images 

#### Description of Task to be completed?
With this feature, **Admins** and the **normal users** should be able to upload images 

- **Using a Helper**
Upload helper can be used in any endpoint
1. make sure you use `form-data` in your form
2. Import the helper `import { upload } from '../helpers';` and use `const image = await upload(req);` in your controller
3. Import **multer** middleware `import { multerUploads } from '../../middlewares/multer';` and use it in your **route**

Expected result format:

```
{
    "image": {
        "info": {
            "image_version": "image_version",
            "image_id": "image_id",
            "image_format": "image_format"
        },
        "image": {
            "original": "image_url,
            "thumbnail": "image_url",
            "square": "image_url",
            "circle": "image_url"
        }
    }
}
```

- **Using an endpoint**

`POST /api/v1/upload` : Upload image to the server

#### How should this be manually tested?
- After cloning the repo, cd into it and type `npm test`
- It should be tested using POSTMAN

#### Any background context you want to provide?
Before testing, ask `API SECRET` and `API KEY`
Upload image on Cloudinary CDN

#### What are the relevant pivotal tracker stories?
[#166325993](https://www.pivotaltracker.com/story/show/166325993)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by e-liyai** [comment]
@djallas I believe when an image is uploaded it is done in relation to an article. this means the article has to have a way to get the URL of the image when it loads. You don't capture that here.

**Comment by djallas** [comment]
@e-liyai I am going to work on that, I will let you know ASAP.

File: src/helpers/upload/upload.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { cloudinaryConfig, uploader } from '../../config/cloudinaryConfig';
+import { dataUri } from '../../middlewares/multer';
+
+export default async (req) => {
+  await cloudinaryConfig();
```
[the remark this answers is not in the document]
Comment (djallas) [reply src/helpers/upload/upload.js]: @e-liyai 
`dataUri` help me to convert image to `base64 encoded string`.  that represents the image to be uploaded(I use it on line 9)

File: package.json [hunk 2]
```diff
@@ -21,17 +21,20 @@
...
     "@babel/register": "^7.4.4",
     "@hapi/joi": "^15.0.3",
     "@sendgrid/mail": "^6.4.0",
     "bcryptjs": "^2.4.3",
     "body-parser": "^1.19.0",
     "chance": "^1.0.18",
     "cheke": "^1.0.3",
+    "cloudinary": "^1.14.0",
     "cors": "^2.8.5",
+    "d": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (djallas) [reply package.json]: the package is no longer needed.

**Pull request merged**